burn-core/src/nn/loss/cross_entropy.rs

```diff
@@ -1,21 +1,107 @@
-use core::marker::PhantomData;
+use crate as burn;
 
-use burn_tensor::{activation, backend::Backend, Bool, Int, Tensor};
+use crate::{config::Config, module::Module};
+use burn_tensor::activation::{log_softmax, softmax};
+use burn_tensor::{backend::Backend, Bool, Int, Tensor};
+
+/// Configuration to create a [Cross-entropy loss](CrossEntropyLoss).
+#[derive(Config)]
+pub struct CrossEntropyLossConfig {
+    /// Create padded cross entropy.
+    ///
+    /// Prevents pad tokens from impacting loss calculation.
+    pad_tokens: Option<Vec<usize>>,
+
+    /// Create weighted cross-entropy.
+    ///
+    /// The loss of a specific sample will simply be given by: weight[y] * log(p(x)) * 1,
+    ///
+    /// # Pre-conditions
+    ///   - The order of the weight vector should correspond to the label integer assignment.
+    ///   - Targets assigned negative Int's will not be allowed.
+    weights: Option<Vec<f32>>,
+
+    /// Create cross-entropy with label smoothing.
+    ///
+    /// Hard labels {0, 1} will be changed to y_smoothed = y(1 - a) + a / nr_classes.
+    /// Alpha = 0 would be the same as default.
+    smoothing: Option<f32>,
+
+    /// Create binary cross-entrop.
+    ///
+    /// Will reduce computation by interpolating secondary label.  
+    /// Should only be given labels Tensor with {0, 1}.
+    /// If weights are also used, only two should be given.     
+    /// Cannot use pad_token at the same time.
+    #[config(default = false)]
+    binary: bool,
```

Comment:
I think we should support binary cross entropy with another module. The forward pass should have a different signature (only tensors of rank 1), the config options won't support padding and the weights config could be `[usize; 2]`. What do your think?